cmd/kubeadm/app/phases/apiconfig/clusterroles.go

```diff
@@ -17,41 +17,72 @@ limitations under the License.
 package apiconfig
 
 import (
-	"fmt"
-
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
 	"k8s.io/kubernetes/cmd/kubeadm/app/master"
+	"k8s.io/kubernetes/pkg/api/v1"
 	rbac "k8s.io/kubernetes/pkg/apis/rbac/v1beta1"
 	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
 )
 
-// CreateBootstrapRBACClusterRole grants the system:node-bootstrapper role to the group we created the bootstrap credential with
-func CreateBootstrapRBACClusterRole(clientset *clientset.Clientset) error {
-	clusterRoleBinding := rbac.ClusterRoleBinding{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "kubeadm:kubelet-bootstrap",
-		},
-		RoleRef: rbac.RoleRef{
-			APIGroup: "rbac.authorization.k8s.io",
-			Kind:     "ClusterRole",
-			Name:     "system:node-bootstrapper",
-		},
-		Subjects: []rbac.Subject{
-			{Kind: "Group", Name: master.KubeletBootstrapGroup},
-		},
+const (
+	// TODO: This role should eventually be a system:-prefixed, automatically bootstrapped ClusterRole
+	KubeDNSClusterRoleName = "kubeadm:kube-dns"
+
+	KubeProxyClusterRoleName        = "system:node-proxier"
+	NodeBootstrapperClusterRoleName = "system:node-bootstrapper"
+
+	// Constants
+	clusterRoleKind    = "ClusterRole"
+	serviceAccountKind = "ServiceAccount"
+	rbacAPIGroup       = "rbac.authorization.k8s.io"
+)
+
+// TODO: Are there any unit tests that could be made for this file other than duplicating all values and logic in a separate file?
+
+// CreateRBACRules creates the essential RBAC rules for a minimally set-up cluster
+func CreateRBACRules(clientset *clientset.Clientset) error {
+	// Create the ClusterRoles we need for our RBAC rules
+	if err := CreateClusterRoles(clientset); err != nil {
+		return err
 	}
-	if _, err := clientset.Rbac().ClusterRoleBindings().Create(&clusterRoleBinding); err != nil {
+
+	// Create the CreateClusterRoleBindings we need for our RBAC rules
+	if err := CreateClusterRoleBindings(clientset); err != nil {
 		return err
 	}
-	fmt.Println("[apiconfig] Created node bootstrapper RBAC rules")
```

Comment:
That's totally true. Is that something you could look into? How we could establish a clean logging framework or something between the different and independent phases?